### PR TITLE
fix issues with QIR in cost function and update expected results

### PR DESCRIPTION
**Pull request by cqc-melf**
# Description

Please summarise the changes.

# Related issues

Solves https://github.com/CQCL/pytket-quantinuum/issues/510
and Solves https://github.com/CQCL/pytket-quantinuum/issues/509 

# Checklist

- [x] I have performed a self-review of my code.
- [ ] I have commented hard-to-understand parts of my code.
- [ ] I have made corresponding changes to the public API documentation.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have updated the changelog with any user-facing changes.

### CODE REVIEW

File: pytket/extensions/quantinuum/backends/quantinuum.py [hunk 1]
```diff
@@ -1078,6 +1076,12 @@ def process_circuits(
                             ),
                         )
                     ).decode("utf-8")
+
+                    for name, count in Counter(
```
Comment (cqc-melf) [inline pytket/extensions/quantinuum/backends/quantinuum.py:1080]: qasm only excludes unused scratch bits, I have now decided to exclude all of them here for QIR. I am not sure if that is the right thing to do, but I would assume that no users are interested in seeing this?
Open to discussions.

File: tests/integration/backend_test.py [hunk 2]
```diff
@@ -1106,12 +1106,22 @@ def test_old_handle(
     assert (r0.get_shots() == r1.get_shots()).all()
 
 
+@pytest.mark.parametrize(
```
Comment (cqc-alec) [inline tests/integration/backend_test.py:1109]: Should we parametrize the other failing test as well?

**Review by cqc-alec — APPROVED** [review]
I rebased #508 onto this branch and the only failure was a timeout, so happy with the fixes.

File: pytket/extensions/quantinuum/backends/quantinuum.py (hunk 1 above)
Comment (cqc-alec) [reply pytket/extensions/quantinuum/backends/quantinuum.py:1080]: Maybe QASM should exclude all of them as well?

File: tests/integration/backend_test.py (hunk 2 above)
Comment (cqc-melf) [reply tests/integration/backend_test.py:1118]: done

File: tests/integration/backend_test.py [hunk 3]
```diff
@@ -834,9 +834,18 @@ def test_wasm(
...
+    [
+        Language.QASM,
+        Language.QIR,
+        Language.PQIR,
+    ],
+)
 @pytest.mark.timeout(120)
 def test_wasm_costs(
     authenticated_quum_backend_qa: QuantinuumBackend,
+    language: Language,
```
Comment (cqc-alec) [inline tests/integration/backend_test.py:848]: Should use this in the call to `cost()`?

Comment (cqc-melf) [reply tests/integration/backend_test.py:848]: Yes, done

**Review by cqc-alec — APPROVED** [review]

**Pull request merged**